lib/chef/knife/bmcs_server_list.rb

```diff
@@ -25,10 +26,42 @@ def run
         options = {}
         options[:limit] = config[:limit] if config[:limit]
 
+        servers = {}
+
         response = compute_client.list_instances(compartment_id, options)
 
-        display_list(response, ['Display Name', 'ID', 'State']) do |item|
-          [item.display_name, item.id, item.lifecycle_state]
+        if config[:format] == 'summary' && response.data
+          # build a hash of hashs, keyed off of server_id
+          response.data.each do |server|
+            vnics = compute_client.list_vnic_attachments(compartment_id, instance_id: server.id)
+            vnics.data.each do |vnic|
+              servers[vnic.instance_id] = {}
+              servers[vnic.instance_id]['vnic_id'] = vnic.vnic_id
+              if vnic.lifecycle_state == 'ATTACHED'
+                begin
+                  vnic_info = network_client.get_vnic(vnic.vnic_id, {})
+                rescue OracleBMC::Errors::ServiceError
+                  servers[vnic.instance_id]['private_ip'] = '(none)'
+                  servers[vnic.instance_id]['public_ip'] = '(none)'
+                else
+                  servers[vnic.instance_id]['private_ip'] = vnic_info.data.private_ip
```

Comment:
Once you get an address, break the loop - both to prevent the address being overwritten with an empty value, and to prevent unnecessary calls.

Actually, even better behavior would be to look for the newly added isPrimary property on VNIC, https://docs.us-phoenix-1.oraclecloud.com/tools/ruby/latest/OracleBMC/Core/Models/Vnic.html#is_primary-instance_method, and have a preference for showing that one. I think that can be added with a later PR and issue though - we'll need to bump the SDK dependency to make use of it.